server/channels/api4/job.go

```diff
@@ -126,7 +126,7 @@ func downloadJob(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
```

Comment:
L103 also needs to be changed to `c.App.ExportFileReader` right?

Given that it's so easy to make mistakes like this, wondering we can add a test for this?

Reply:
@agnivade Yes -- interestingly though, this is legacy code, so we've actually never supported downloading from the dedicated file store. Hah.

I fully support locking this down with a test, but after spending 30 minutes on it, I think we're not set up to selectively mock just the reader calls from `App` -- i.e. to test whether the `File/ZipReader` or the `ExportFile/ZipReader` are called. Unless you can think of a way to do it, we're kind of stuck. Limitations of Go mocking I think... ?  Open to suggestions.

Reply:
That's fine. Let's leave it. Also wondering if we can simply rip out this legacy code? Why is it still there?

Reply:
@agnivade Backwards compatibility -- it's here for download links that were made with versions < 10.5

Reply:
Gotcha. I thought that code was ancient.